spec/lib/manageiq/api/common/entitlements_spec.rb

```diff
@@ -0,0 +1,29 @@
+describe ManageIQ::API::Common::Entitlement do
+  let(:encoded) { encoded_user_hash }
+  let(:request_good) do
+    { :headers => { 'x-rh-identity' => encoded }, :original_url => 'whatever' }
+  end
+
+  around do |example|
+    ManageIQ::API::Common::Request.with_request(request_good) { example.call }
+  end
+
+  context "entitlement getter methods" do
+    let(:entitlement_keys) { %w[insights openshift hybrid_cloud smart_management] }
+    let(:bad_entitlement)  { %w[fred barney type] }
+    let(:entitlement)      { ManageIQ::API::Common::Request.current.entitlement }
+    let(:other_user)       { default_user_hash }
+
+    it "returns values for entitlement methods" do
+      entitlement_keys.each do |key|
+        expect(entitlement.respond_to?("#{key}?")).to be_truthy
```

Comment:
spec-wise, this can be hard to find issues when something goes wrong because rspec will stop on the failed key, but it won't show you which key is failing.  It's usually better to loop the other way around...

```ruby
entitlement_keys.each do |key|
  it "returns values for #{key} entitlement" do
    expect(entitlement.respond_to?("#{key}?")).to be_truthy
  end
end
```